sklearn/ensemble/_hist_gradient_boosting/_predictor_bitset.pyx

```diff
@@ -0,0 +1,104 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: language_level=3
+# cython: nonecheck=False
+# distutils: language=c++
+
+from ._bitset cimport in_bitset
+from .common cimport BITSET_INNER_DTYPE_C
+from libc.limits cimport CHAR_BIT
+from libcpp.vector cimport vector
+
+
+cdef inline unsigned char in_vec_bitset(vector[BITSET_INNER_DTYPE_C]& bitset,
+                                        int value) nogil:
+    cdef:
+        unsigned int i1 = value // 32
+        unsigned int i2 = value % 32
+
+    if bitset.size() < i1 + 1:
+        return 0
+    return (bitset[i1] >> i2) & 1
+
+
+cdef inline void insert_vec_bitset(vector[BITSET_INNER_DTYPE_C]& bitset,  # OUT
+                                   int value) nogil:
+    cdef:
+        unsigned int i1 = value // 32
+        unsigned int i2 = value % 32
+
+    if bitset.size() < i1 + 1:
+        bitset.resize(i1 + 1, 0)
```

Comment:
Each time the vector is resized, a new memory section needs to be allocated and the data is copied there. Will this be an issue here?

Note that Python lists and C++ vectors are implemented by doubling the allocated size each time (maybe another ratio is used).

Should we rather use `push_back` (equivalent of append) in order to let C++ handle the resizing logic?

That being said, don't we know the maximum size of the array in advance? There should be no more than 256 bits in this array.